### ISSUE TITLE
problem about GridFeatureNetwork code

### ISSUE THREAD

**Issue by verigle** [davidnvq/grit#22]
```
class GridFeatureNetwork(nn.Module):

    def __init__(
        self,
        n_layers,
        pad_idx,
        d_in=1024,
        d_model=512,
        n_heads=8,
        d_ff=2048,
        dropout=0.1,
        attn_dropout=0.0,
        attention_module=None,
        **kwargs,
    ):
        super().__init__()
        self.fc = nn.Linear(d_in, d_model)
        self.dropout = nn.Dropout(p=dropout)
        self.layer_norm = nn.LayerNorm(d_model)
        self.layers = nn.ModuleList([
            TransformerLayer(
                d_model,
                n_heads,
                d_ff,
                dropout,
                attn_dropout=attn_dropout,
                attention_module=attention_module,
                **kwargs,
            ) for _ in range(n_layers)
        ])

    def forward(self, input, attention_mask=None, attention_weights=None):
        out = F.relu(self.fc(input))
        out = self.dropout(out)
        out = self.layer_norm(out)

        if attention_mask is None:
            attention_mask = (torch.sum(out, dim=-1) == self.padding_idx)
            attention_mask = repeat(attention_mask, 'B N -> B 1 1 N')  # [B Head Nq N]

        outs = []
        for l in self.layers:
            out = l(out, out, out, attention_mask, attention_weights)
            outs.append(out.unsqueeze(1))

        outs = torch.cat(outs, 1)
        return outs, attention_mask
```

there are some problem about GridFeatureNetwork for file of models.caption.grid_net.py 
'GridFeatureNetwork' object has no attribute 'padding_idx'， however `self.padding_idx` was used in  `attention_mask = (torch.sum(out, dim=-1) == self.padding_idx)`
i

**Comment by davidnvq** [comment]
Thank you for pointing out this issue. 
You are right. Indeed, I had one line in the old code: `self.padding_idx = pad_idx`. It was missing during the refactoring.
